File: Day1/SonarSweepWind.java
import java.util.Scanner;
import java.io.File;
import java.io.FileNotFoundException;
public class SonarSweepWind {
    public static void main(String[] args) throws FileNotFoundException{
        int value[] = {0,0,0,0};
        int amount = 0;
        Scanner sc = new Scanner(new File("input.txt"));
        for(int i = 1; i < 4; i++){
            value[i] = sc.nextInt();
        }
        while(sc.hasNext()){
            value = nextValue(sc, value);
            if(value[0] < value[3]){
                amount++;
            }
        }
        System.out.println(amount);
    }
    private static int[] nextValue(Scanner sc, int[] value){
        //Move
        for(int i = 0; i<3; i++){
            value[i] = value[i+1];
        }
        value[3] = sc.nextInt();
        return value;
    }
}

File: Day1/SonarSweepWind.py
def nextValue(value, line):    
    for x in range(3):
        value[x] = value[x+1]
    value[3] = int (line)
    return value
    
file = open('input.txt', 'r')
value = [0,0,0,0]
amount = 0

for x in range(1, 4):
    value[x] = int (file.readline())

for line in file:
    value = nextValue(value, line)
    if(value[0] + value[1] + value[2] < value[1] + value[2] + value[3]):
        amount += 1
print(amount)
file.close()